examples/all-clusters-app/all-clusters-common/src/WhmManufacturer.cpp

```diff
@@ -0,0 +1,286 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <WhmInstance.h>
+#include <WhmManufacturer.h>
+#include <water-heater-mode.h>
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/clusters/water-heater-management-server/WaterHeaterManagementTestEventTriggerHandler.h>
+#include <app/clusters/water-heater-management-server/water-heater-management-server.h>
+
+using namespace chip;
+using namespace chip::app::Clusters::WaterHeaterManagement;
+
+using Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace WaterHeaterManagement {
+
+CHIP_ERROR WhmManufacturer::Init()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmManufacturer()->GetWhmDelegate();
+    if (dg == nullptr)
+    {
+        ChipLogError(AppServer, "WhmDelegate is not initialized");
+        return CHIP_ERROR_UNINITIALIZED;
+    }
+
+    dg->SetHeaterTypes(BitMask<WaterHeaterTypeBitmap>(WaterHeaterTypeBitmap::kImmersionElement1));
+    dg->SetHeatDemand(BitMask<WaterHeaterDemandBitmap>(WaterHeaterDemandBitmap::kImmersionElement1));
+    dg->SetEstimatedHeatRequired(10000);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR WhmManufacturer::Shutdown()
+{
+    return CHIP_NO_ERROR;
+}
+
+BitMask<WaterHeaterDemandBitmap> WhmManufacturer::DetermineHeatingSources()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmManufacturer()->GetWhmDelegate();
+    if (dg == nullptr)
+    {
+        ChipLogError(AppServer, "WhmDelegate is not initialized");
+        return BitMask<WaterHeaterDemandBitmap>(0);
+    }
+
+    // A list of valid heaterTypes
+    uint8_t waterHeaterTypeValues[] = {
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kImmersionElement1),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kImmersionElement2),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kHeatPump),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kBoiler),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kOther),
+    };
+
+    // The corresponding list of valid headerDemands
+    uint8_t waterHeaterDemandValues[] = {
+        static_cast<uint8_t>(WaterHeaterDemandBitmap::kImmersionElement1),
```

Comment:
Again, why not an array of `WaterHeaterDemandBitmap`?

Reply:
There is a pending spec PR which will get rid of WaterHeaterTypeBitmap and WaterHeaterDemandBitmap and replace with WaterHeaterHeatSourceBitmap - so propose leaving this until that is reworked.

Added this to https://github.com/project-chip/connectedhomeip/issues/34565 follow up issue.